R/rlistings.R

```diff
@@ -1,47 +1,45 @@
+#'   `listing_df` to be modified.
+#' @param key_cols (`character`)\cr vector of names of columns which should be treated as *key columns*
 #'   when rendering the listing. Key columns allow you to group repeat occurrences.
-#' @param disp_cols character or NULL. Names of non-key columns which should be displayed when
-#'   the listing is rendered. Defaults to all columns of `df` not named in `key_cols` or
+#' @param disp_cols (`character` or `NULL`)\cr vector of names of non-key columns which should be
+#'   displayed when the listing is rendered. Defaults to all columns of `df` not named in `key_cols` or
 #'   `non_disp_cols`.
-#' @param non_disp_cols character or NULL. Names of non-key columns to be excluded as display
-#'   columns. All other non-key columns are then treated as display columns. Invalid if
-#'   `disp_cols` is non-NULL.
-#' @param unique_rows logical(1). Should only unique rows be included in the listing. Defaults to `FALSE`.
-#' @param default_formatting list. A named list of default column format configurations to apply when rendering the
-#'   listing. Each name-value pair consists of a name corresponding to a data class (or "numeric" for all unspecified
-#'   numeric classes) and a value of type `fmt_config` with the format configuration that should be implemented for
-#'   columns of that class. If named element "all" is included in the list, this configuration will be used for all
-#'   data classes not specified. Objects of type `fmt_config` can take 3 arguments: `format`, `na_str`, and `align`.
-#' @param col_formatting list. A named list of custom column formatting configurations to apply to specific columns
-#'   when rendering the listing. Each name-value pair consists of a name corresponding to a column name and a value of
-#'   type `fmt_config` with the formatting configuration that should be implemented for that column. Objects of type
-#'   `fmt_config` can take 3 arguments: `format`, `na_str`, and `align`. Defaults to `NULL`.
-#' @param main_title character(1) or NULL. The main title for the listing, or
-#'   `NULL` (the default). Must be length 1 non-NULL.
-#' @param subtitles character or NULL. A vector of subtitle(s) for the
-#'   listing, or `NULL` (the default).
-#' @param main_footer character or NULL. A vector of main footer lines
-#'   for the listing, or `NULL` (the default).
-#' @param prov_footer character or NULL. A vector of provenance strings
-#'   for the listing, or `NULL` (the default). Each string element is placed on a new line.
-#' @param vec any. A column vector from a `listing_df` to be annotated as a key column.
+#' @param non_disp_cols (`character` or `NULL`)\cr vector of names of non-key columns to be excluded as display
+#'   columns. All other non-key columns are treated as display columns. Ignored if `disp_cols` is non-`NULL`.
+#' @param unique_rows (`logical(1)`)\cr whether only unique rows should be included in the listing. Defaults to `FALSE`.
```

Comment:
If we are trying to get to {tern} style I would use `flag` for `logical(1)` and `string` for `character(1)`. I do not remember the default for "any" though